src/main/java/com/uid2/shared/auth/OperatorKey.java

```diff
@@ -77,12 +85,13 @@ public boolean equals(Object o) {
                 && this.name.equals(b.name)
                 && this.contact.equals(b.contact)
                 && this.protocol.equals(b.protocol)
+                && this.roles.equals(this.roles)
                 && this.created == b.created;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(key, name, contact, protocol, created);
+        return Objects.hash(key, name, contact, protocol, created, roles);
```

Comment:
i tend to think while roles/siteid are OperatorKey's member variables, they can in theory change values  (public <=> private for example) unlike name/contact/key so i don't think roles should be used to generate hash code? any thoughts?

Reply:
Ah, yes I agree, I will take it out